compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -447,7 +447,9 @@ void addMultiTilingExpertPassPipeline(OpPassManager &passManager,
   nestedModulePM.addNestedPass<func::FuncOp>(
       createLLVMCPUSplitReductionPass(clEnableReassociateFpReductions));
   nestedModulePM.addNestedPass<func::FuncOp>(
-      createLLVMCPUTilePass(numLevels - 1));
+      createLLVMCPUTilePass(numLevels - 1, /*reductionOnly=*/true));
```

Comment:
This ordering might be problematic..... Once you tile the reduction, there is no real opportunity for tile and fuse... Why do we need the tile and fuse after this layer?

Reply:
The order is intended. This is for the last level of tiling, i.e., the tiling level right before vectorization. We want to tile ops individually. We firstly tile the reduction ops, and then handle the consumer ops. There are no differences between tile and TileAndFuse if we have a single consumer op. They will just tile the consumer op. But it's important if there is a consumer ops chain, .g., reduction + broadcast + tensor.pack/pad ops.  I want to tile and fuse `broadcast + pack` op in this case.

Reply:
(maybe I should add a comment, that would help others to understand that it's intended)

Reply:
(Now that I get it) THis is what I meant https://github.com/hanhanW/iree/blob/multi-lowering-config/compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp#LL438C1-L438C1

Instead of invoking the pass for all the dimensions, couldn't we just add a loop that takes care of the parallel/reduction dimensions using the enums?

Reply:
I don't get the idea.. I thought using enums will be done in your multi-level tiling PR.

What I did here is creating different passes for the last level tiling (which intends to be vector level). What do you mean by "invoking the pass for all the dimensions"? Am I doing it now?

The reduction tiling still take the same config (e.g., [0, 0, 16]) for tiling. It only tile the reduction loop.

TileAndFuse for consumer ops are tricky. They reuse the same configuration. I'm still working on adding multi configs support. I see this is a transition state and it is a incremental change towards multi lowering configs.

Reply:
Oh, we only have enums for workgroups right now it seems? https://github.com/openxla/iree/blob/main/compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.h#L22

I was hoping we could do the dimension selection/filtering on the caller side, as we do in [hanhanW/iree@multi-lowering-config/compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp#LL438C1-L438C1](https://github.com/hanhanW/iree/blob/multi-lowering-config/compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp?rgh-link-date=2023-05-30T19%3A06%3A23Z#LL438C1-L438C1), so that LLVMCPUTile/TileAndFuse only have to worry about the tiling and not the filtering. If that is not possible, please go ahead with this as is.

Reply:
Yes, we only have enums for workgroups right now. The passes themselves only worry about the tiling. The filtering is handled by us, i.e., when we create the pass.